src/platform_impl/android/mod.rs

```diff
@@ -739,11 +739,11 @@ impl From<u64> for WindowId {
 }
 
 #[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
-pub struct DeviceId;
+pub struct DeviceId(i32);
 
 impl DeviceId {
     pub const fn dummy() -> Self {
-        DeviceId
+        DeviceId(std::i32::MIN)
```

Comment:
it doesn't matter much but I think we could use zero here.

We don't have to worry about whether the ID overlaps with a real device ID since this `dummy()` API is specifically documented to only be applicable for testing:

```rust
    /// The only guarantee made about the return value of this function is that
    /// it will always be equal to itself and to future values returned by this function.
    /// No other guarantees are made. This may be equal to a real `DeviceId`.
```

Reply:
I think `0` makes sense - have made that change in a squashed commit!